lightfm/lightfm.py

```diff
@@ -601,20 +617,29 @@ def _run_epoch(self, item_features, user_features, interactions,
 
         # Call the estimation routines.
         if loss == 'warp':
-            fit_warp(CSRMatrix(item_features),
-                     CSRMatrix(user_features),
-                     positives_lookup,
-                     interactions.row,
-                     interactions.col,
-                     interactions.data,
-                     sample_weight,
-                     shuffle_indices,
-                     lightfm_data,
-                     self.learning_rate,
-                     self.item_alpha,
-                     self.user_alpha,
-                     num_threads,
-                     self.random_state)
+
+            # Cycle over scopes
+            # TODO catch scope=None and match_indices=None
```

Comment:
What happens if `scope is None` and `matc_indices is NOT None`?

Reply:
good question :) I would say scope is None will set match_indices to None even if match_indices are passed, because if scope is None we should revert to the original sampling in lightfm